entitlement/src/main/java/org/killbill/billing/entitlement/engine/core/DefaultEventsStream.java

```diff
@@ -88,11 +91,10 @@ public DefaultEventsStream(final ImmutableAccountData account,
                                final SubscriptionBase subscription,
                                final Collection<SubscriptionBase> allSubscriptionsForBundle,
                                @Nullable final Integer defaultBillCycleDayLocal,
-                               final InternalTenantContext contextWithValidAccountRecordId, final DateTime utcNow) {
+                               final InternalTenantContext contextWithValidAccountRecordId, final DateTime utcNow, final boolean includeDeletedEvents) {
```

Comment:
I am not super excited about passing both the `blockingStates` (unclear what those are unless you look at how it's been called) and the `includeDeletedEvents` flag.

Ideally, we would want the impl. to be as symmetrical as what we have done in `DefaultSubscriptionBase`, but I understand it's not easy as 'events' get computed inside `DefaultSubscriptionBase` and here they are computed outside (so I understand why you did that).

Perhaps:
1. We pass both `blockingStates` and `blockingStatesWithDeletedEvents` 
2. We compute the value of `includeDeletedEvents ` based on the lists we receive - see comment below.
3. We have to define the semantics of what blockingStatesWithDeletedEvents contains when caller was called with `includeDeletedEvents=false` (In `DefaultSubscriptionBase`, I think this is indeed an empty list), hence my comment below for the `getBlockingStates()` method.

Reply:
I've now made changes as per my understanding of the above, please let me know if it still does not look good.